### PR TITLE
Update Dockerfile to match Devbox Cloud

### PR DESCRIPTION
**Pull request by Lagoja**
## Summary

TSIA -- moves our generated dockerfile to a `debain:stable-slim` base

## How was it tested?

`docker build .` on an example project

### CODE REVIEW

File: internal/impl/tmpl/Dockerfile.dockerignore.tmpl
Comment (mohsenari) [inline internal/impl/tmpl/Dockerfile.dockerignore.tmpl:1]: @Lagoja This is the wrong file to update.
This is dockerignore. You should update `devcontainerDockerfile.tmpl`

**Review by mohsenari — CHANGES_REQUESTED** [review]
Wrong file is updated.

File: internal/impl/tmpl/devcontainerDockerfile.tmpl [hunk 1]
```diff
@@ -1,30 +1,30 @@
...
 
-# Setting up devbox user
-ENV DEVBOX_USER=devbox
-RUN adduser -h /home/$DEVBOX_USER -D -s /bin/bash $DEVBOX_USER
-RUN addgroup sudo
-RUN addgroup $DEVBOX_USER sudo
-RUN echo " $DEVBOX_USER      ALL=(ALL:ALL) NOPASSWD: ALL" >> /etc/sudoers
+# Step 1: Installing dependencies
+RUN apt-get update
+RUN apt-get -y install bash binutils git xz-utils wget sudo
```
Comment (hezhizhen) [inline internal/impl/tmpl/devcontainerDockerfile.tmpl:5]: How about choose a base image that contains these tools (or some of them)?

File: internal/impl/tmpl/devcontainerDockerfile.tmpl [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
 
-# installing devbox
-RUN wget --quiet --output-document=/dev/stdout https://get.jetpack.io/devbox | bash -s -- -f
+# Step 3: Installing devbox
+RUN wget --quiet --output-document=/dev/stdout https://get.jetpack.io/devbox   | bash -s -- -f
 RUN chown -R "${DEVBOX_USER}:${DEVBOX_USER}" /usr/local/bin/devbox
 
-# nix installer script
-RUN wget --quiet --output-document=/dev/stdout https://nixos.org/nix/install | sh -s -- --no-daemon
+# Step 4: Install Nix
```
Comment (hezhizhen) [inline internal/impl/tmpl/devcontainerDockerfile.tmpl:19]: Use `Installing` to follow the pattern

File: internal/impl/tmpl/devcontainerDockerfile.tmpl [hunk 3]
```diff
@@ -1,30 +1,30 @@
...
+RUN wget --output-document=/dev/stdout https://nixos.org/nix/install | sh -s -- --no-daemon
 RUN . ~/.nix-profile/etc/profile.d/nix.sh
 # updating PATH
 ENV PATH="/home/${DEVBOX_USER}/.nix-profile/bin:/home/${DEVBOX_USER}/.devbox/nix/profile/default/bin:${PATH}"
 
+# Step 5: Install your devbox project
```
Comment (hezhizhen) [inline internal/impl/tmpl/devcontainerDockerfile.tmpl:25]: `Installing`

File: internal/impl/tmpl/devcontainerDockerfile.tmpl [hunk 4]
```diff
@@ -1,30 +1,30 @@
...
+RUN apt-get update
+RUN apt-get -y install bash binutils git xz-utils wget sudo
 
-# installing dependencies
-RUN apk add --no-cache bash binutils git libstdc++ xz sudo
+# Step 2: Setting up devbox user
+ENV DEVBOX_USER=devbox
+RUN adduser $DEVBOX_USER
+RUN usermod -aG sudo $DEVBOX_USER
+RUN echo " devbox      ALL=(ALL:ALL) NOPASSWD: ALL" | sudo tee /etc/sudoers.d/$DEVBOX_USER
```
Comment (hezhizhen) [inline internal/impl/tmpl/devcontainerDockerfile.tmpl:11]: Why do we need some many spaces between `devbox` and `ALL`?

File: internal/impl/tmpl/devcontainerDockerfile.tmpl (hunk 1 above)
Comment (mohsenari) [reply internal/impl/tmpl/devcontainerDockerfile.tmpl:5]: @hezhizhen The motivation for using this base image is that, we use the same base image in devbox cloud instances. To keep the image size small, it's better to use a minimal image and install the tools you need.

**Review by mohsenari — APPROVED** [review]
There are some small linting updates that I can take care. Otherwise, looks good.

File: internal/impl/tmpl/devcontainerDockerfile.tmpl (hunk 1 above)
Comment (hezhizhen) [reply internal/impl/tmpl/devcontainerDockerfile.tmpl:5]: I see.

**Pull request merged**